error when compiled using gcc-aarch64-none-elf

error message:  
core/src/impl/Kokkos_HostBarrier.cpp:75:7: error: 'nanosleep' was not declared in this scope
core/src/impl/Kokkos_HostBarrier.cpp:77:7: error: 'sched_yield' was not declared in this scope
though time.h and sched.h are included.

--- a/core/src/impl/Kokkos_HostBarrier.cpp
+++ b/core/src/impl/Kokkos_HostBarrier.cpp
@@ -61,7 +61,6 @@ namespace Impl {
 
 void HostBarrier::impl_backoff_wait_until_equal(
     int* ptr, const int v, const bool active_wait) noexcept {
-#if !defined(_WIN32)
   unsigned count = 0u;
 
   while (!test_equal(ptr, v)) {
@@ -87,18 +86,6 @@ void HostBarrier::impl_backoff_wait_until_equal(
 #endif
 #endif
   }
-#else  // _WIN32
-  while (!test_equal(ptr, v)) {
-#if defined(KOKKOS_ENABLE_ASM)
-    for (int j = 0; j < num_nops; ++j) {
-      __asm__ __volatile__("nop\n");
-    }
-    __asm__ __volatile__("pause\n" ::: "memory");
-#endif
-  }
-#endif
-  // printf("W: %d\n", count);
 }
-
 }  // namespace Impl
 }  // namespace Kokkos

--- a/core/src/impl/Kokkos_HostBarrier.cpp
+++ b/core/src/impl/Kokkos_HostBarrier.cpp
@@ -49,10 +49,8 @@
 
 #include <impl/Kokkos_HostBarrier.hpp>
 
-#if !defined(_WIN32)
-#include <sched.h>
-#include <time.h>
-#else
+#include <thread>
+#if defined(_WIN32)
 #include <process.h>
 #include <winsock2.h>
 #include <windows.h>
@@ -64,17 +62,15 @@ namespace Impl {
 void HostBarrier::impl_backoff_wait_until_equal(
     int* ptr, const int v, const bool active_wait) noexcept {
 #if !defined(_WIN32)
-  timespec req;
-  req.tv_sec     = 0;
   unsigned count = 0u;
 
   while (!test_equal(ptr, v)) {
     const int c = ::Kokkos::log2(++count);
     if (!active_wait || c > log2_iterations_till_sleep) {
-      req.tv_nsec = c < 16 ? 256 * c : 4096;
-      nanosleep(&req, nullptr);
+      std::this_thread::sleep_for(
+          std::chrono::nanoseconds(c < 16 ? 256 * c : 4096));
     } else if (c > log2_iterations_till_yield) {
-      sched_yield();
+      std::this_thread::yield();
     }
 #if defined(KOKKOS_ENABLE_ASM)
 #if defined(__PPC64__)

--- a/core/src/impl/Kokkos_Spinwait.cpp
+++ b/core/src/impl/Kokkos_Spinwait.cpp
@@ -49,12 +49,8 @@
 #include <impl/Kokkos_Spinwait.hpp>
 #include <impl/Kokkos_BitOps.hpp>
 
-#if defined(KOKKOS_ENABLE_STDTHREAD) || defined(_WIN32)
 #include <thread>
-#elif !defined(_WIN32)
-#include <sched.h>
-#include <time.h>
-#else
+#if defined(_WIN32)
 #include <process.h>
 #include <winsock2.h>
 #include <windows.h>
@@ -73,28 +69,14 @@ void host_thread_yield(const uint32_t i, const WaitMode mode) {
 
   if (WaitMode::ROOT != mode) {
     if (sleep_limit < i) {
-      // Attempt to put the thread to sleep for 'c' milliseconds
-
-#if defined(KOKKOS_ENABLE_STDTHREAD) || defined(_WIN32)
-      auto start = std::chrono::high_resolution_clock::now();
+      // Attempt to put the thread to sleep for 'c' microseconds
       std::this_thread::yield();
-      std::this_thread::sleep_until(start + std::chrono::nanoseconds(c * 1000));
-#else
-      timespec req;
-      req.tv_sec  = 0;
-      req.tv_nsec = 1000 * c;
-      nanosleep(&req, nullptr);
-#endif
+      std::this_thread::sleep_for(std::chrono::microseconds(c));
     }
 
     else if (mode == WaitMode::PASSIVE || yield_limit < i) {
       // Attempt to yield thread resources to runtime
-
-#if defined(KOKKOS_ENABLE_STDTHREAD) || defined(_WIN32)
       std::this_thread::yield();
-#else
-      sched_yield();
-#endif
     }
 #if defined(KOKKOS_ENABLE_ASM)
 

